### DISCUSSION TITLE
Access relationship columns in newly instantiated models

### DISCUSSION THREAD

**Discussion by elliotdes** [sqlalchemy/sqlalchemy#11223] [Usage Questions] [answered]
Hi,

if I have the following models:

```python
class League(DeclarativeBase):
    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str]

class Team(DeclarativeBase):
    id: Mapped[int] = mapped_column(primary_key=True)
    league_id: Mapped[int] = mapped_column(ForeignKey("league.id"))
    name: Mapped[str]
    league: Mapped["League"] = relationship()
```
    
  and I already have a league in my database with an id of 1.

How can I access this league from a newly instantiated `Team` that has not yet been commited to the database?

For example:

```python
new_team = Team(league_id=1, name="my_team")
# this equals None, I want it to be League(id=1, name="my_league")
new_team.league
```

**Comment by zzzeek** [comment, marked as answer]
this is not a "mainstream" pattern and I would be very careful as to what the rationale is for this.   However, there are special use functions that allow this.  see https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.enable_relationship_loading  .
